doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,324 @@
+- [Configuration and management](#configuration-and-management)
+- [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+  - [CLI](#cli)
+  - [Yang model](#yang-model)
+  - [Config DB Enhancements](#config-db-enhancements)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictionslimitations)
+- [Testing Requirements/Design](#testing-requirementsdesign)
+  - [System Test Cases](#system-test-cases)
+
+# Revision History
+|  Rev  | Date       |   Author    | Change Description                |
+| :---: | :--------- | :---------: | --------------------------------- |
+|  0.1  | 08/13/2021 |  Jimi Chen  | Initial version                   |
+|  0.2  | 08/30/2021 |  Jimi Chen  | Update community review comments  |
+
+# Scope 
+This HLD extends SONiC to support static anycast gateway
+
+# Abbreviations
+| Abbreviations | Description            |
+|---------------|------------------------|
+| SAG           | Static Anycast Gateway |
+
+# Overview
+Anycast gateway feature for EVPN/VXLAN is a default gateway addressing mechanism that enables the use of the same gateway IP address across all the leaf switches that are part of a VXLAN network.  
+This ensures that every leaf switch can function as the default gateway for the workloads directly connected to it.  
+The feature failitates flexible workload placement, host mobility and optimal traffic forwarding across the VXLAN fabric.  
+
+This feature is used together with VxLAN EVPN usually, it can also be used individually.  
+In common use case, SAG announces its IP and MAC towards the host facing Ethernet ports but not towards the fabric.  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric facing side.
+
+# Requirements
+User can configure global virtual MAC address to apply on VLAN interfaces.   
+Also provides a per VLAN interface's knob to determine if VLAN interface use this global virtual MAC address or system CPU MAC address.
+
+
+# Architecture Design
+The overall SONiC architecture will not be changed and no new sub-modules will be introduced.
+
+# High-Level Design
+## sonic-swss-common
+sonic-swss-common's schema will be updated to include **SAG** define for the new table name.
+
+## sonic-swss
+sonic-swss's Intfs Orch and Intf Mgr will be updated to include a new handler for static anycast gateway configuration, and also handles new enable/disable field for VLAN interface in original handler.
+
+The IP address configured on the VLAN interface is not changed.  
+The VLAN interface will use static anycast gateway MAC address to replace CPU MAC address if static anycast gateway MAC address is specified and it's enabled on the VLAN interface.  
+The enable/disable knob on the VLAN interface can let user to determine to use CPU MAC or static anycast gateway MAC address.
+
+The static anycast gateway on the VLAN interface will be disabled by default.
```

Comment:
How about autogenerated IPv6 LL on SAG enabled vlan. Kernel will update the EUI-64(based on mac) IPv6 auto-ll only after netdev flap. This needs to be handled explicitly by flapping vlan netdev when mac of vlan changes from system-mac to SAG MAC or SAG MAC update case.

Reply:
You're right, I'll add description for handling IPv6 auto-ll after netdev flap in the updated HLD.

Reply:
@preetham-singh , reference to IPv6 link-local enhancement HLD, the generated MAC IPv6 link-local route to me will be updated by calling RouteOrch's API when the VLAN MAC address changed in the updated HLD.
Please help to review it, thanks.